src/main/java/frc/robot/util/Trigger.java

```diff
@@ -0,0 +1,25 @@
+package frc.robot.util;
+
+import frc.robot.actions.Action;
+import com.google.gson.Gson;
+
+public class Trigger {
+    public int buttonID;
+    public Action action;
+    private Gson gson;
+
+    public Trigger (int buttonID, Action action) {
+        this.buttonID = buttonID;
+        this.action = action;
+        gson = new Gson();
+    }
+
+    public void loop()
+    {
+        if(Context.robotController.driverJoystick.getButtonPressed(buttonID)){
+            String seralized = gson.toJson(this.action);
```

Comment:
It isn't obvious what a google library for converting stuff to JSON is doing here. This a place where comments would be nice to have.